tests/pricing_models/test_calc_out_given_in.py

```diff
@@ -0,0 +1,1611 @@
+                        #                         = 102.50671833648673
+                        without_fee_or_slippage=102.50671833648673,
+                        # We want to solve for the amount of bonds out given the
+                        # amount of shares coming in, so we set up the problem as:
+                        #
+                        #   k = (c / mu) * (mu * (z + d_z)) ** (1 - tau) + (2 * y + c * z - d_y') ** (1 - tau)
+                        #     = 100_100 ** (1 - tau) + (300_000 - d_y') ** (1 - tau)
+                        #
+                        # Solving for d_y, we get the following calculation:
+                        #
+                        #   d_y' = 300_000 - (k - 100_100 ** (1 - tau) ** (1 / (1 - tau))
+                        #       = 102.50516899477225
+                        #
+                        # Note that this is slightly smaller than the without slippage value
+                        without_fee=102.50516899477225,
+                        # fee = 0.01 * (p - 1) * 100 = 0.02506718336486724
+                        fee=0.02506718336486724,
+                        # with_fee = d_y' - fee
+                        #          = 102.50516899477225 - 0.02506718336486724
+                        #          = 102.48010181140738
+                        with_fee=102.48010181140738,
+                    ),
+                    hyperdrive=TestResultCalcOutGivenInSuccess(
+                        # spot_price = (1 / p) * c * delta_z * t + c * delta_z * (1 - t)
+                        # spot_price = (1 / 0.9517182274304707) * 1 * 100.0 * 0.5 + 1 * 100.0 * 0.5
+                        without_fee_or_slippage=102.53655815229496,
+                        # yield space equation (for hyperdrive multiply d_z by t):
+                        # k = (c / mu) * (mu * (z + d_z))**(1 - tau) + (2y + cz - d_y')**(1 - tau)
+                        # dy' = 2y + c*z - (k - (c / mu) * (mu * (z + d_z))**(1 - tau))**(1 / (1 - tau))
+                        #
+                        # note: use delta_z * t to phase the curve part out, and add delta_z * (1 - t)
+                        # to phase the flat part in over the length of the term:
+                        #
+                        # dy' = 2*y + c*z - (k - (c / u) * (u * (z + delta_z*t))**(1 - tau_full))**(1 / (1 - tau_full))
+                        #       + c * delta_z * (1 - t)
+                        # dy' = 199900.0 + 100050.0 - (229443.78725685307 - 1.0 * (1 * (100050.0 + 100.0*0.5))**(0.9549283119368059))**(1.0471990279267966)
+                        #       + 1 * 100.0 * 0.5
+                        without_fee=102.5357591957436,
+                        # fee = ((1 / p) - 1) * phi * c * d_z
+                        # fee = (1.050731163045899 - 1) * 0.01 * 1 * 100.0
+                        fee=0.025365581522949543,
+                        with_fee=102.51039361422065,
+                    ),
```

Comment:
Your math and explanations look good. The fact that this is different than the example above in all cases except where `time_remaining = 365 days` indicates that we have an issue with time in Hyperdrive. We discussed this previously, and I think we need to remedy the situation before we give the sims to maker.